File: apps/backend/app/schemas/points_wallet.py
# app/schemas/points_wallet.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import List
from uuid import UUID
from enum import Enum

class PointsBalance(BaseModel):
    balance: int
    model_config = ConfigDict()

class PointsOperation(BaseModel):
    points: int

    model_config = ConfigDict()

class TransactionType(str, Enum):
    CREDIT = "credit"
    DEBIT  = "debit"

class PointsTransaction(BaseModel):
    id: UUID      
    type: TransactionType
    amount: int
    description: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PaginatedPointsTransactions(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[PointsTransaction]

    model_config = ConfigDict(from_attributes=True)

File: apps/portal-company/src/types/pointsWallet.ts
// src/types/pointsWallet.ts
export interface PointsBalance {
  balance: number;
}

/**
 * Tipo de operação na carteira de pontos
 */
export type TransactionType = 'credit' | 'debit';

/**
 * Item de transação de crédito/débito da carteira de pontos
 */
export interface PointsTransaction {
  id: string;
  type: TransactionType;
  amount: number;
  description?: string;
  created_at: string; // ISO date
}

/**
 * Resposta paginada do extrato de transações de pontos da empresa logada
 */
export interface PaginatedPointsTransactions {
  total: number;
  skip: number;
  limit: number;
  items: PointsTransaction[];
}
